### PR TITLE
add UI and logic to header-new

### PR DESCRIPTION
**Pull request by nvolodymyr**

### CODE REVIEW

File: src/app/component/core/header-new/header-new.component.html [hunk 1]
```diff
@@ -1,24 +1,27 @@
...
   <div class="main-container">
-    <div class="header">
+    <div class="header" >
       <div class="logo">
         <a href="#">
           <picture>
           <source media="(max-width: 576px)"
                   srcset="assets/img/logo_S.png">
-          <img src="assets/img/logo_M.png" alt="logo">
+             <img src="assets/img/logo_M.png" alt="logo">
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.html:9]: Please write attributes in separate rows

File: src/app/component/core/header-new/header-new.component.html [hunk 2]
```diff
@@ -56,29 +59,118 @@
...
-              <li>    <div id="switcher-wrapper">
-                <img id="arrow" [src]="arrow" alt="v">
-                <select [(ngModel)]="language" (change)="changeCurrentLanguage()" id="language-switcher">
-                  <option class="language-option" [ngValue]="'uk'">Ua</option>
-                  <option class="language-option" [ngValue]="'ru'">Ru</option>
-                  <option class="language-option" [ngValue]="'en'">En</option>
+               <li class="search"
+                   [class.destroy]="onToggleBurgerMenu">
+               <a  href="#">
+                 <img src="assets/img/icon/search.png" alt="search">
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.html:102]: Please write attributes in separate rows

File: src/app/component/core/header-new/header-new.component.html [hunk 3]
```diff
@@ -56,29 +59,118 @@
...
+            <a (click)="openSingInWindow()">
+              {{ "top-bar.sign-in" | translate }}
+            </a>
+          </ng-template>
+          <ng-template #userNotLoggedIn>
+            <div id="user-avatar-wrapper">
+             <ul [class.add-shadow]="dropdownVisible">
+               <li class="tertiary-global-button">
+                 <a (click)="toggleDropdown()">Nadiia Steblivets
+                 <img class="arrow" [src]="arrowGreen"
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.html:135]: Same as line 102

File: src/app/component/core/header-new/header-new.component.ts [hunk 4]
```diff
@@ -14,22 +13,24 @@ import { HabitStatisticService } from 'src/app/service/habit-statistic/habit-sta
 import { filter } from 'rxjs/operators';
 import { LanguageService } from '../../../i18n/language.service';
 import { Language } from '../../../i18n/Language';
+import {SignInNewComponent} from '../../auth/sign-in-new/sign-in-new.component';
+import {NewSignUpComponent} from '../../auth/new-sign-up/new-sign-up.component';
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:17]: General: add space between brackets in imports

File: src/app/component/core/header-new/header-new.component.ts [hunk 5]
```diff
@@ -53,6 +54,13 @@ export class HeaderNewComponent implements OnInit {
       .subscribe(this.assignData.bind(this));
     this.userRole = this.jwtService.getUserRole();
     this.language = this.languageService.getCurrentLanguage();
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:56]: Please create a separate fucntion and use it here.

File: src/app/component/core/header-new/header-new.component.ts [hunk 6]
```diff
@@ -70,4 +78,54 @@ export class HeaderNewComponent implements OnInit {
     this.userId = userId;
     this.isLoggedIn = true;
   }
+
+  toggleDropdown(): void {
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:82]: Please, add access modificator

File: src/app/component/core/header-new/header-new.component.ts [hunk 7]
```diff
@@ -70,4 +78,54 @@ export class HeaderNewComponent implements OnInit {
...
+
+  private openSignUpWindow(): void {
+    this.dialog.open(NewSignUpComponent, {
+      hasBackdrop: true,
+      closeOnNavigation: true,
+      panelClass: 'custom-dialog-container',
+    });
+  }
+
+  openDialog(): void {
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:102]: Same as line 82

File: src/app/component/core/header-new/header-new.component.ts [hunk 8]
```diff
@@ -70,4 +78,54 @@ export class HeaderNewComponent implements OnInit {
...
+    this.dropdownVisible = false;
+    const dialogRef = this.dialog.open(FavoritePlaceComponent, {
+      width: '700px'
+    });
+    dialogRef.afterClosed().subscribe(() => {
+      this.favoritePlaceService.getFavoritePlaces();
+    });
+  }
+
+  openSettingDialog(): void {
```
Comment (RomanDeveloperAcc) [inline src/app/component/core/header-new/header-new.component.ts:112]: Same as line 82

File: src/app/component/core/header-new/header-new.component.html (hunk 1 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.html:9]: +

File: src/app/component/core/header-new/header-new.component.html (hunk 2 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.html:102]: +

File: src/app/component/core/header-new/header-new.component.html (hunk 3 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.html:135]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 4 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:17]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 6 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:82]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 7 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:102]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 8 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:112]: +

File: src/app/component/core/header-new/header-new.component.ts (hunk 5 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.ts:52]: fix

File: src/app/app.component.html [hunk 9]
```diff
@@ -1,10 +1,10 @@
 <div class="flash-message">
```
Comment (nabermi2020) [inline src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: Margins should be the same as on the mockup (Breakpoint >1200). Please, also check it on the other breakpoints

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: Please, fix paddings according to the mockup

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: Active item in the header should be green

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: Please, check dimensions for hamburger menu and also paddings between lines.

Comment (nabermi2020) [reply src/app/app.component.html:1]: image

Comment (nabermi2020) [reply src/app/app.component.html:1]: Add 'border-radius' for every line

File: src/app/component/core/header-new/header-new.component.html [hunk 10]
```diff
@@ -56,29 +55,123 @@
...
+                          <li class="mobile-vis">
+                              <a (click)="signOut()">
+                            {{ "nav-bar.sign-out" | translate }}
+                             </a>
+                          </li>
+                        </ng-template>
+                        <li class="mobile-vis"
+                            *ngIf="!isLoggedIn">
+                          <div class="create-button secondary-global-button">
+                            <span (click)="openSignUpWindow()">Sign up</span>
```
Comment (nabermi2020) [inline src/app/component/core/header-new/header-new.component.html:89]: Please, use localization

Comment (nabermi2020) [reply src/app/component/core/header-new/header-new.component.html:89]: image

File: src/app/app.component.html (hunk 9 above)
Comment (nvolodymyr) [reply src/app/app.component.html:1]: fix

File: src/app/component/core/header-new/header-new.component.html (hunk 10 above)
Comment (nvolodymyr) [reply src/app/component/core/header-new/header-new.component.html:89]: fix

**Pull request merged**